### PR TITLE
speedtest-rs: init at 0.1.4

### PR DESCRIPTION
**Pull request by GaetanLepage**
###### Description of changes

Added package `speedtest-rs`, a command line internet speedtest tool written in rust.
Homepage: https://github.com/nelsonjchen/speedtest-rs

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 218429` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>speedtest-rs</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 218429` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>speedtest-rs</li>
  </ul>
</details>

File: pkgs/tools/networking/speedtest-rs/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "speedtest-rs";
+  version = "0.1.4";
+
+  src = fetchFromGitHub {
+    owner = "nelsonjchen";
+    repo = pname;
+    rev = "v${version}";
```
Comment (fabaff) [inline pkgs/tools/networking/speedtest-rs/default.nix:15]: Sometime the automated update processes are running into issue if a tag/branch confusion is present.

```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/tools/networking/speedtest-rs/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  buildInputs = [ openssl ];
+
+  nativeBuildInputs = [ pkg-config ];
+
+  cargoSha256 = "sha256-4TJEM+oMjx/aaZgY2Y679pYFTdEWWFpWDYrK/o2b5UM=";
+
+  meta = with lib; {
+    description = "Command line internet speedtest tool written in rust";
+    homepage = "https://github.com/nelsonjchen/speedtest-rs";
```
Comment (fabaff) [inline pkgs/tools/networking/speedtest-rs/default.nix:27]: Tools can benefit from a changelog entry as some are able to add those information of the commit message or the PR description.

```suggestion
    homepage = "https://github.com/nelsonjchen/speedtest-rs";
    changelog = "https://github.com/nelsonjchen/speedtest-rs/blob/v${version}/CHANGELOG.md";
```

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 218429` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>speedtest-rs</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/218429#issuecomment-1449061472)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-218429-1]$ ./results/speedtest-rs/bin/speedtest-rs 
Retrieving speedtest.net configuration...
Retrieving speedtest.net server list...
Testing from Hetzner Online GmbH (88.198.53.140)...
Selecting best server based on latency...
Hosted by Deutsche Telekom (Munich) [0.07 km]: 10.40 ms
Testing download speed........................................................^C
[...]
```

Looks good to me.

**Pull request merged**